### ISSUE TITLE
setInterval on appInBackground?

### ISSUE THREAD

**Issue by gldstrrbt** [angeloraso/capacitor-plugin-background-mode#1]
First, thanks so much for creating this plugin. It's the only background plugin I can seem to get working at the moment.

I've been looking around for a Capacitor background plugin that allows for the running of a setInterval function, or at least runs a given js function periodically. There's the transistor software plugin that logcat does show to run every 15 minutes, but doesn't appear to run the js (react) script I've set for it.

From what I've tested within the last hour or so, it appears that your app can run a one time script on appInBackground. Is it possible to run a setInterval function as well? I've added a setInterval function but it doesn't seem to be running. 

Thank you in advance.
-----------------------------
Here's what I tried:
`		    BackgroundMode.addListener("appInBackground", ()=>{
		    	setInterval(()=>{
			    	LocalNotifications.schedule({
				  		notifications: [
						    {
					      		title: "Please work",
					      		body: "componentdidmount: backgroundmode: appInBackground: setinterval",
					      		id: new Date.getTime(),
					      		schedule: { at: new Date(Date.now() + 1000 * 5) },
					      		sound: null,
					      		attachments: null,
					      		actionTypeId: "",
					      		extra: null
				    		}
				  		]
					});
		    	},20000);
		    })`

**Comment by angeloraso** [comment]
- Hello gldstrrbt! Sorry for the delay. I just managed to free myself to analyze your problem.
- I would assume that what you want to do should work. The plugin tricks the operating system into thinking that it is still in the foreground. This unfortunately consumes battery power but should allow you to execute any code in javascript
- As soon as I can, I'll try your solution. If it doesn't work I'll try to offer you a new one

**Comment by angeloraso** [comment]
- Hi @gldstrrbt. I just tested your problem with a simple app with the latest version of capacitor-plugin-background-mode. At least in Android Studio emulator it works successfully. I put a setInterval (1000ms) after the "appInBackground" event with a simple console.log. Everything works correctly.
- Perhaps with a physical device it does not happen in the same way
- Anyway, if this problem persists, I would recommend creating a separate plugin that takes care of start a service (Android) and running your asynchronous tasks once the "appInBackground" event is fired
- I hope I've helped
